test/functional/bundleadd/add-alias-basic.bats

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env bats
+
+# Author: <author name>
+# Email: <author email>
+
+load "../testlib"
+
+global_setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -n test-bundle1 -f /foo/test-file1 "$TEST_NAME"
+	create_bundle -n alias-bundle1 -f /foo/alias-file1 "$TEST_NAME"
+	create_bundle -n alias-bundle2 -f /foo/alias-file2 "$TEST_NAME"
+	create_bundle -n alias-bundle3 -f /foo/alias-file3 "$TEST_NAME"
+}
+
+test_setup() {
+
+	return
+
+}
+
+test_teardown() {
+
+	remove_bundle -L "$TEST_NAME"/web-dir/10/Manifest.test-bundle1
+	remove_bundle -L "$TEST_NAME"/web-dir/10/Manifest.alias-bundle1
+	remove_bundle -L "$TEST_NAME"/web-dir/10/Manifest.alias-bundle2
+	remove_bundle -L "$TEST_NAME"/web-dir/10/Manifest.alias-bundle3
+	sudo sh -c "rm -rf "$TARGETDIR"/usr/share/defaults/swupd/alias.d/"
+	sudo sh -c "rm -rf "$TARGETDIR"/etc/swupd/alias.d/"
+	clean_state_dir "$TEST_NAME"
+
+}
+
+global_teardown() {
+
+	destroy_test_environment "$TEST_NAME"
+
+}
+
+@test "ADD032: Add bundle without alias" {
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS test-bundle1"
+
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Starting download of remaining update content. This may take a while...
+		.
+		Finishing download of update content...
+		Installing bundle(s) files...
+		.
+		Calling post-update helper scripts.
+		Successfully installed 1 bundle
+	EOM
+	)
+	assert_is_output --identical "$expected_output"
+	assert_file_exists "$TARGETDIR"/foo/test-file1
+	assert_file_exists "$TARGETDIR"/usr/share/clear/bundles/test-bundle1
+
+}
+
+@test "ADD033: Add bundle with system alias" {
+
+	sudo sh -c "mkdir -p $TARGETDIR/usr/share/defaults/swupd/alias.d/"
+	sudo sh -c "echo -e 'alias1\talias-bundle1' > $TARGETDIR/usr/share/defaults/swupd/alias.d/a1"
+        run cat "$TARGETDIR/usr/share/defaults/swupd/alias.d/a1"
+        eo=$(cat <<-EOM
+alias1	alias-bundle1
+EOM
+)
+        assert_is_output --identical "$eo"
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS alias1"
+
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Starting download of remaining update content. This may take a while...
+		.
+		Finishing download of update content...
+		Installing bundle(s) files...
+		.
+		Calling post-update helper scripts.
+		Successfully installed 1 bundle
+	EOM
+	)
+	assert_is_output --identical "$expected_output"
+	assert_file_exists "$TARGETDIR"/foo/alias-file1
+	assert_file_exists "$TARGETDIR"/usr/share/clear/bundles/alias-bundle1
+
+}
+
+@test "ADD034: Add bundle with user alias" {
+
+	sudo sh -c "mkdir -p $TARGETDIR/etc/swupd/alias.d/"
+        run ls "$TARGETDIR/etc/swupd/"
+        assert_is_output --identical "alias.d"
```

Comment:
it would be better if you use _assert_in_output_ instead of _assert_is_output_ since you are only interested into making sure that dir is there, because if we add some other content to that /etc/swupd directory your test is going to fail

Reply:
That was only to help me debug the test failing later on.